api/proto/agent/actions.proto

```diff
@@ -15,7 +15,8 @@ message ExecutionContext {
 message ActionRequest {
     uint32 action_id = 1;
     ExecutionContext context = 2;
-    repeated string commands = 3;
+    string repo_url = 3;
+    repeated string commands = 4;
```

Comment:
This is just a note, since we are in a very small project ;
but good practice (I should say, necessary practices) when adding a field in a protobuf file, is to not modify the order of the other strings.

This is FYI, but keeping the numbering (1 to 3) and simply adding a new field (`repo_url`) as a fourth field allows retro-compatibility and forward-compatibility, meaning no changes to code won't break the API interface.

Do as you wish (as it is not so important now), but in later stages I suggest keeping `commands` as the third field and adding repo_url as the fourth.